pkg/kubectl/sorting_printer.go

```diff
@@ -185,16 +186,37 @@ func isLess(i, j reflect.Value) (bool, error) {
 
 func (r *RuntimeSort) Less(i, j int) bool {
 	iObj := r.objs[i]
+	iElem := reflect.ValueOf(iObj).Elem()
 	jObj := r.objs[j]
+	jElem := reflect.ValueOf(jObj).Elem()
+	if r.field == "" && iElem.Type().Name() == "ReplicaSet" && jElem.Type().Name() == "ReplicaSet" {
```

Comment:
This is the part that made me describe the change as "gross". I know very little about this whole conversion business; what's a better approach? I'm assuming it would require further plumbing upstream from this code.